src/simple_rules.ml

```diff
@@ -19,7 +19,21 @@ let user_rule sctx ?extra_bindings ~dir ~scope (rule : Rule.t) =
   let targets : SC.Action.targets =
     match rule.targets with
     | Infer -> Infer
-    | Static fns -> Static (List.map fns ~f:(Path.relative dir))
+    | Static fns ->
+      let f fn =
+        let loc = String_with_vars.loc fn in
+        List.map ~f:(function
+          | "." | ".." ->
+            Errors.fail loc "'.' and '..' are not valid filenames"
+          | fn ->
+            if Filename.dirname fn <> Filename.current_dir_name then
+              Errors.fail
+                loc
+                "%s does not denote a file in the current directory" fn;
+            Path.relative ~error_loc:loc dir fn
+        ) (SC.expand_vars_string_list sctx ~scope ~dir fn)
+      in
+      Static (List.concat_map ~f fns)
```

Comment:
This will work, but it's a bit weird to stringify the paths and then turn them back to paths again.

It would be best to just expand things to `Value.t` and then pattern match on it in the `List.map`. You can handle paths just by letting them through, strings can have the same validation + conversion, and the rest we should error out on.

This is what I meant in my original comment. So if it's not clear, I can make the little tweak myself.

Reply:
But I would still need to check that `Path` values denote a file in the current directory, right? In which am not sure this really would simplify much.

Reply:
Thanks, I added a commit with this suggestion.